### PR TITLE
feat: add types

### PR DESCRIPTION
**Pull request by azan-n**

### CODE REVIEW

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -37,7 +38,19 @@ jobs:
...
+  # Compile TS files to JS
+  compile:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - run: npm ci
+      - run: npm run compile
```
Comment (adnan-kamili) [inline .github/workflows/npm-publish.yml:51]: npm run build

File: package.json [hunk 2]
```diff
@@ -2,8 +2,10 @@
   "name": "@cryptlex/lexactivator",
   "version": "3.20.4",
   "description": "LexActivator API wrapper for Node.js.",
-  "main": "index.js",
+  "main": "lib/index.js",
+  "types": "lib/index.d.ts",
   "scripts": {
+    "compile": "tsc",
```
Comment (adnan-kamili) [inline package.json:8]: build

File: src/lexactivator.ts [hunk 3]
```diff
@@ -43,17 +43,19 @@ export class ProductVersionFeatureFlag {
...
 }
 
+export type ActivationModes = 'online' | 'offline';
+
 /**
  *  @class ActivationMode
  *  @constructor
- *  @property {TODO} initialMode initial mode of activation.
- *  @property {} currentmode current mode of activation.
+ *  @property {ActivationModes} initialMode initial mode of activation.
```
Comment (adnan-kamili) [inline src/lexactivator.ts:51]: ActivationModes or ActivationMode?

Comment (azan-n) [reply src/lexactivator.ts:51]: the type declaration will conflict with the existing ActivationMode class. Please suggest other names if required.

**Review by adnan-kamili — APPROVED** [review]

**Pull request merged**